webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -506,6 +533,56 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
       this.openDirectory.next(results.directory);
     });
   }
+
+
+  doSavingDataset(context: ProjectContext, requestUrl: string, _activeFile: ProjectContext, results: any, isUntagged: boolean,
+    _observer: Observer<void>, _observable: Observable<void>) {
+
+      this.snackBar.open(`Attempting to save Dataset ${results}`, 'Close', { duration: MessageDuration.Short, panelClass: 'center' });
+
+      /* Send the HTTP PUT request to the server
+    * to save the file.
+    */
+    /*  POST ... */
+    this.ngHttp.post(requestUrl, _activeFile.model.contents).subscribe(r => {
+
+        /* It was a new file, we
+        * can set the new fileName. */
+        // if (results && !isUntagged) {
+        //   _activeFile.name = results.fileName;
+        //   _activeFile.model.name = results.fileName;
+        //   _activeFile.model.fileName = results.fileName;
+        //   _activeFile.model.encoding = this.getIntEncoding(results.encoding);
+        //   _activeFile.model.path = results.directory;
+        //   _activeFile.temp = false;
+        // }
+        /* This will probably need to be changed
+        * for the sake of accessibility.
+        */
+        this.snackBar.open(`Dataset ${results} has been saved!`, 'Close', { duration: MessageDuration.Short, panelClass: 'center' });
+
+        /* Send buffer saved event */
+        this.bufferSaved.next({ buffer: _activeFile.model.contents, file: _activeFile.model.name });
+        let fileList = this.openFileList.getValue()
+          .map(file => {
+            if (file.id === context.id) {
+              file.changed = false;
+            }
+            return file;
+          });
+        this.openFileList.next(fileList);
+        this.openDirectory.next(results.directory);
+        if (_observer != null) { _observer.next(null); }
+    }, e => {
+        let error = e.json().error;
+
+        /* This will probably need to be changed
+        * for the sake of accessibility.
+        */
+        this.snackBar.open(`Dataset ${results} could not be saved.  ${error}`, 'Close', { duration: MessageDuration.Medium, panelClass: 'center' });
```

Comment:
I can't tell if this is the whole error object JSON-ified. If it is, that's not good for the user, we'd just want to print the error.code & error.msg (or sometimes error.body or something) but if it's not, then we're all set